classloader-leak-prevention/classloader-leak-prevention-core/src/test/java/se/jiderhamn/classloader/leak/prevention/cleanup/ReplaceDOMNormalizerSerializerAbortExceptionCleanUpTest.java

```diff
@@ -13,4 +16,12 @@
   public void triggerLeak() throws Exception {
     ReplaceDOMNormalizerSerializerAbortExceptionInitiatorTest.triggerLeak();
   }
+
+    @Override
+    public void triggerLeakWithoutCleanup() throws Exception {
+        // Leak does not occur any more with JDK8+
+        Assume.assumeTrue(JavaVersion.IS_JAVA_1_6 || JavaVersion.IS_JAVA_1_7);
```

Comment:
Would it have made more sense to make the original `triggerLeakWithoutCleanup()` do something like `Assume.assumeTrue(isRelevantForJavaVersion())` with a default implementation of `isRelevantForJavaVersion()` that returns true, and then override that method in the respective tests? Feels like it would be more readable?

Plus, `isRelevantForJavaVersion()` could be named the same for  both CleanUp tests and PreInitiator tests.

Reply:
Indeed.